src/transformers/models/wav2vec2_conformer/modeling_wav2vec2_conformer.py

```diff
@@ -724,13 +724,13 @@ def _apply_rotary_embedding(self, hidden_states, relative_position_embeddings):
         batch_size, sequence_length, hidden_size = hidden_states.size()
         hidden_states = hidden_states.view(batch_size, sequence_length, self.num_heads, self.head_size)
 
-        cos = relative_position_embeddings[0, :sequence_length, ...]
-        sin = relative_position_embeddings[1, :sequence_length, ...]
+        cos = relative_position_embeddings[0, :sequence_length, :]
+        sin = relative_position_embeddings[1, :sequence_length, :]
 
         # rotate hidden_states with rotary embeddings
         hidden_states = hidden_states.transpose(0, 1)
-        rotated_states_begin = hidden_states[..., : self.head_size // 2]
-        rotated_states_end = hidden_states[..., self.head_size // 2 :]
+        rotated_states_begin = hidden_states[:, : self.head_size // 2]
+        rotated_states_end = hidden_states[:, self.head_size // 2 :]
```

Comment:
If `hidden_states` here is the shape `(sequence_length, batch_size, self.num_heads, self.head_size)` then this operation isn't equivalent

```
In [96]: x = torch.arange(1 * 2 * 3 * 4 * 5).reshape(1, 2, 3, 4, 5)

In [97]: x[..., :2].shape
Out[97]: torch.Size([1, 2, 3, 4, 2])

In [98]: x[:, :2].shape
Out[98]: torch.Size([1, 2, 3, 4, 5])
```

Reply:
Yeah indeed, it depends on the dimension and it's too hacky to do all of them in transformers just for a bug in pytorch...

Thanks Amy! We will wait for a patch on PyTorch side then!